### PR TITLE
Allow organization admins to interact with invitations

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
This PR allows organization admins to interact with pending and rejected invitations. It relies on this webservice PR https://github.com/dockstore/dockstore/pull/5243. 

The UI closely follows this [mock-up](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/602c28cebb6cf41a28cc204e), but there's some deviations based on what I thought made the most sense for us.
- Pending invitations don't have a `Resend Invite` button because I don't think that makes sense for us. Instead, they have the `Edit` button so an admin can edit the invitation role
- There was no mock-up for rejected invites, but I made them look like pending invitations, except instead of an `Edit` button, it has a `Resend Invite` button 

Functionality changes:
- Organization admins can view pending and rejected invitations.
- For pending invitations, they can:
  - Edit the member role
  - Delete the invitation
- For rejected invitations, they can:
  - Resend the invite (the requested user will see it in their requests tab)
  - Delete the invitation. Rejected invitations do not go away on their own. The admin must delete it to remove the organization user.
- The functionality of accepted members remains the same. Admins can edit their roles or remove them from the organization
- The members count displayed in the Members tab includes pending and rejected invitations for the admin. The public view only shows accepted members
- The members are displayed in one column if the user can edit membership so that the buttons can fit without overlapping in the card. This also matches the mock-up
- Updated the Add User button to Invite a Member to match the mock-up. Also updated the dialog to match this [mock-up](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/602c2802942e481ec7cb7d63).

Other changes:
- Noticed some tests in organization.ts were being skipped, so I removed `.skip` and updated them so they pass
- Remove duplicate function declarations in constants.ts
- Updated some related dialog button texts to match the other re-designed dialogs we have
  - Ex: `No thanks` -> `Cancel`
- Updated the Add Collection button so it matches the mock-up because I also updated the Member's Add User button to match the mock-up
- Made a small change in a CWL workflow file in the db dump because it was causing two tests to fail. It wasn't revealed before because the UI tests were using the 1.13.0 webservice artifact and not what's latest in develop.
  - This workflow was causing a `Unprocessable Entity: Unable to parse CWL workflow` when getting the tools json table. The fix was to fix the indentation of one line in the workflow file

I attached some screenshots and videos to show the UI and the functionality. The videos don't work for me in Firefox so if the same happens to you, try Chrome.

Organization admin view before:
Screenshot 2022-11-17 at 13-18-23 Dockstore Organization

Organization admin view after:
image

Admin interacting with pending invitation:

https://user-images.githubusercontent.com/25287123/202487052-9f92c777-94eb-4dc1-8520-470c0e0b6fca.mp4

Admin resending invite for rejected invitation:

https://user-images.githubusercontent.com/25287123/202485656-add5a7b4-8383-4687-842f-78be720fedb7.mp4

Admin deleting rejected invitation:

https://user-images.githubusercontent.com/25287123/202485728-9421a355-2b8a-44b3-943b-98a582a48fd6.mp4

Admin interacting with accepted invitation:

https://user-images.githubusercontent.com/25287123/202485788-097419f9-2545-4cbd-8555-ca300421012d.mp4

I'll update the CircleCI jar to one off the develop branch after I merge the webservice PR

**Review Instructions**
- Create an organization and invite users to it. Verify that users who haven't accepted the invite yet are displayed in the Members tab and that you can edit the member's role and delete the invitation.
- Invite a user then have that user reject the invite. Verify that the user shows up in the Members tab and that you can resend an invitation to the user. This should update the invitation from Rejected Invitation to Pending Invitation.

**Issue**
https://github.com/dockstore/dockstore/issues/4226 
[DOCK-1781](https://ucsc-cgl.atlassian.net/browse/DOCK-1781)

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [ ] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
  - Might be better to wait for the new dashboard My Organizations to be implemented before updating the docs

### CODE REVIEW

**Comment by denis-yuen** [comment]
> Made a small change in a CWL workflow file in the db dump because it was causing two tests to fail. It wasn't revealed before because the UI tests were using the 1.13.0 webservice artifact and not what's latest in develop.

Neat, FYI @svonworl

**Review by denis-yuen — APPROVED** [review]
Nice!
BTW, videos work on Firefox on my desktop

**Review by svonworl — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]
Kudos on the videos!

Looking at this code, and in regards to my comment on your web PR, I see that we have always had the [UI filter members](https://github.com/dockstore/dockstore-ui2/blob/b5250087e1c93a5bf655aa286817787d5357ded8/src/app/organizations/state/organization-members.service.ts#L108). 

So, is it worth changing? Our orgs are currently so lightweight that it probably doesn't matter that much, but on the other hand, I wouldn't necessarily want it made public that I rejected an org invite. As a comparison, I don't think you see org invites/rejections on GitHub unless you're an admin.

I know we're not showing it in the UI currently, but technically that info shouldn't be available via the API either.

**Comment by kathy-t** [comment]
> Looking at this code, and in regards to my comment on your web PR, I see that we have always had the [UI filter members](https://github.com/dockstore/dockstore-ui2/blob/b5250087e1c93a5bf655aa286817787d5357ded8/src/app/organizations/state/organization-members.service.ts#L108).

> I know we're not showing it in the UI currently, but technically that info shouldn't be available via the API either.

I changed the webservice PR so that only organization admins can view all members. This allowed me to remove the member filtering in the UI. Re-requesting your review

**Review by coverbeck — APPROVED** [review]

File: cypress/integration/smokeTests/sharedTests/waf.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ describe('Test AWS Managed Common Rule Set', () => {
   // clients shouldn't read or run. Example patterns include extensions like .log and .ini.
   it('Try a URL that would break if RestrictedExtensions_URIPATH were turned on', () => {
     const longUrl =
-      '/api/ga4gh/trs/v2/tools/%23workflow%2Fgithub.com%2Fnf-core%2Fhlatyping/versions/1.1.2/plain-NFL/descriptor//nextflow.config';
```
Comment (denis-yuen) [inline cypress/integration/smokeTests/sharedTests/waf.ts:76]: This is a little odd if you needed it to pass. The type here should be case insensitive and allow both underscore and dashes

For example
```
~/dockstore-tool-registry-service-schemas$ curl -X 'GET'   'http://phoenix:4200/api/ga4gh/trs/v2/tools/registry.hub.docker.com%2Fsevenbridges%2Fpcawg_bwa_sbg_modified/versions/1.0/plain-cwl/descriptor'   -H 'accept: application/json' | md5sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8503    0  8503    0     0   414k      0 --:--:-- --:--:-- --:--:--  415k
ba0205d2450045caf65f6fd3e00d4d9c  -
~/dockstore-tool-registry-service-schemas$ curl -X 'GET'   'http://phoenix:4200/api/ga4gh/trs/v2/tools/registry.hub.docker.com%2Fsevenbridges%2Fpcawg_bwa_sbg_modified/versions/1.0/PLAIN-CWL/descriptor'   -H 'accept: application/json' | md5sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8503    0  8503    0     0   406k      0 --:--:-- --:--:-- --:--:--  415k
ba0205d2450045caf65f6fd3e00d4d9c  -
~/dockstore-tool-registry-service-schemas$ curl -X 'GET'   'http://phoenix:4200/api/ga4gh/trs/v2/tools/registry.hub.docker.com%2Fsevenbridges%2Fpcawg_bwa_sbg_modified/versions/1.0/PLAIN_CWL/descriptor'   -H 'accept: application/json' | md5sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8503    0  8503    0     0   491k      0 --:--:-- --:--:-- --:--:--  518k
ba0205d2450045caf65f6fd3e00d4d9c  -
```

File: src/app/container/info-tab/info-tab.component.ts [hunk 2]
```diff
@@ -193,7 +193,7 @@ export class InfoTabComponent extends Base implements OnInit, OnChanges {
   getTRSLink(path: string, versionName: string, descriptorType: ToolDescriptor.TypeEnum, relativePath: string): string {
     return (
       `${Dockstore.API_URI}${ga4ghPath}/tools/${encodeURIComponent(path)}` +
-      `/versions/${encodeURIComponent(versionName)}/plain-` +
```
Comment (denis-yuen) [inline src/app/container/info-tab/info-tab.component.ts:196]: Ditto, this should not be necessary

File: src/app/shared/descriptor-type-compat.service.ts [hunk 3]
```diff
@@ -68,4 +73,34 @@ export class DescriptorTypeCompatService {
     }
     return extendedUnknownDescriptor.plainTRS;
   }
+
+  /**
+   * Converts the ToolDescriptor.TypeEnum to ToolVersion.DescriptorTypeEnum
+   */
+  public toolDescriptorTypeEnumToToolVersionDescriptorTypeEnum(typeEnum: ToolDescriptor.TypeEnum): ToolVersion.DescriptorTypeEnum | null {
```
Comment (denis-yuen) [inline src/app/shared/descriptor-type-compat.service.ts:80]: Ah, this part makes sense.

**Review by denis-yuen — COMMENTED** [review]
Ok, one part makes sense, one part doesn't and may mean that there is a bigger bug at play

**Review by denis-yuen — APPROVED** [review]
Discussed, case changes are not actually failing when running against the webservice

**Comment by kathy-t** [comment]
> Discussed, case changes are not actually failing when running against the webservice

FYI, I changed references of `plain-` or `PLAIN-` to `PLAIN_` to match the webservice openapi: https://github.com/dockstore/dockstore/blob/86ba4b876c0018cd7aff7a43557869473759536d/dockstore-webservice/src/main/resources/openapi3/openapi.yaml#L2828-L2832

I didn't change it to fix any failures against the webservice

**Pull request merged**